pkg/statsd/statsd.go

```diff
@@ -88,12 +86,6 @@ func socketFactory(metricsAddr string, connPerReader bool) SocketFactory {
 func (s *Server) createStandaloneSink() (gostatsd.PipelineHandler, []gostatsd.Runnable, error) {
 	var runnables []gostatsd.Runnable
 
-	for _, backend := range s.Backends {
```

Comment:
Server receives backends as a parameter - it is not the owner, it didn't create them. So it should not be managing their lifecycle. That is why I moved this up one level into whatever constructs the `Server`. Same applies to `CachedInstances` and cloud providers.